libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/swap_ranges.hpp

```diff
@@ -0,0 +1,355 @@
+//  Copyright (c) 2015-2020 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file parallel/container_algorithms/swap_ranges.hpp
+
+#pragma once
+
+#if defined(DOXYGEN)
+
+namespace hpx { namespace ranges {
+    // clang-format off
+
+    ///////////////////////////////////////////////////////////////////////////
+    /// Exchanges elements between range [first1, last1) and another range
+    /// starting at \a first2.
+    ///
+    /// \note   Complexity: Linear in the distance between \a first1 and \a last1
+    ///
+    /// \tparam InIter1     The type of the first range of iterators to swap
+    ///                     (deduced).
+    /// \tparam Sent1       The type of the first sentinel (deduced). This
+    ///                     sentinel type must be a sentinel for InIter1.
+    /// \tparam InIter2     The type of the second range of iterators to swap
+    ///                     (deduced).
+    /// \tparam Sent2       The type of the second sentinel (deduced). This
+    ///                     sentinel type must be a sentinel for InIter2.
+    ///
+    /// \param first1       Refers to the beginning of the sequence of elements
+    ///                     for the first range.
+    /// \param last1        Refers to sentinel value denoting the end of the
+    ///                     sequence of elements for the first range.
+    /// \param first2       Refers to the beginning of the sequence of elements
+    ///                     for the second range.
+    /// \param last2        Refers to sentinel value denoting the end of the
+    ///                     sequence of elements for the second range.
+    ///
+    /// The swap operations in the parallel \a swap_ranges algorithm
+    /// invoked without an execution policy object  execute in sequential
+    /// order in the calling thread.
+    ///
+    /// \returns  The \a swap_ranges algorithm returns
+    ///           \a util::in_in_result<InIter1, InIter2>.
+    ///           The \a swap_ranges algorithm returns in_in_result with the
+    ///           first element as the iterator to the element past the last
+    ///           element exchanged in range beginning with \a first1 and the
+    ///           second element as the iterator to the element past the last
+    ///           element exchanged in the range beginning with \a first2.
+    ///
+    template <typename InIter1, typename Sent1, typename InIter2,
+        typename Sent2>
+    parallel::util::in_in_result<InIter1, InIter2>
```

Comment:
IIRC, in other places we import `in_in_result` (and friends) into the namespace `hpx::ranges`. We might have to consolidate this, though.

Reply:
Hmmm, we should rather explicitly reference `hpx::ranges::swap_ranges_result` here.